expan/core/statistics.py

```diff
@@ -614,16 +614,72 @@ def compute_p_value(mean1, std1, n1, mean2, std2, n2):
     """
 
     # First, check if we have enough data to do a t-test
-    if min(n1,n2) < 1 or max(n1,n2) < 2:
+    if min(n1, n2) < 1 or max(n1, n2) < 2:
         return np.nan
 
     mean_diff = mean1 - mean2
-    std       = pooled_std(std1, n1, std2, n2)
-    st_error  = std * np.sqrt(1. / n1 + 1. / n2)
-    d_free    = n1 + n2 - 2
+    std = pooled_std(std1, n1, std2, n2)
+    st_error = std * np.sqrt(1. / n1 + 1. / n2)
+    d_free = n1 + n2 - 2
     if st_error == 0.0:
-        t         = np.sign(mean_diff) * 1000
+        t = np.sign(mean_diff) * 1000
     else:
-        t         = mean_diff / st_error
-    p         = stats.t.cdf(-abs(t), df=d_free) * 2
+        t = mean_diff / st_error
+    p = stats.t.cdf(-abs(t), df=d_free) * 2
     return p
+
+
+def chi_square(x, y, min_counts=5):
+    """ Computes chi-square p-value and chi-square test statistics on samples x and y.
+    
+    :param x: sample of the treatment group
+    :type  x: pd.Series or array-like
+    :param y: sample of the control group
+    :type  x: pd.Series or array-like
+    :param min_counts: minimum number of observed and expected frequencies (should be at least 5)
```

Comment:
where does the number `5` come from?

Reply:
"This test is invalid when the observed or expected frequencies in each category are too small. A typical rule is that all of the observed and expected frequencies should be at least 5." - https://docs.scipy.org/doc/scipy-0.16.1/reference/generated/scipy.stats.chisquare.html